pkg/nrtupdater/nrtupdater.go

```diff
@@ -125,6 +140,9 @@ func (te *NRTUpdater) Run(infoChannel <-chan MonitorInfo) chan<- struct{} {
 				if te.args.Oneshot {
 					break
 				}
+				if condChan != nil {
```

Comment:
this is good. A (IMHO) better approach is to provide a helper function like

```
package podreadines

...

func Set(cond v1.PodCondition) {
   if condChan != nil {
     condChan <- cond
  }
}
```
This way:
1. we factor the mechanism to set the condition into the relevant package (the fact we use a goroutine is a implementation detail
2. client code is not concerned about how the setting happens, just signals the intention to set.

Reply:
I see. But the function is not aware of the channel, shall I pass it as well?

Reply:
@fromanirh ^

Reply:
in general, better to hide the channel. So let's pass around a new type:
```
type Injector struct {}

...

func (inj *Injector) Set(cond v1.PodCondition) {}

...
```

Reply:
OK, what about this?
we change `podreadiness` like this
```
func SetCondition(condChan chan<- v1.PodCondition, condType RTEConditionType, condStatus v1.ConditionStatus) {
	if condChan == nil {
		return
	}
	cond := newConditionTemplate(condType, condStatus)
	if condStatus == v1.ConditionFalse {
		switch condType {
		case ResourcesScanned:
			cond.Reason = "ScanFailed"
			cond.Message = "failed to scan pod resources"
		case TopologyUpdated:
			cond.Reason = "UpdateFailed"
			cond.Message = "failed to update noderesourcetopology object"
		}
	}
	condChan <- cond
}

func newConditionTemplate(condType RTEConditionType, status v1.ConditionStatus) (condition v1.PodCondition) {
	return v1.PodCondition{
		Type:               v1.PodConditionType(condType),
		Status:             status,
		LastTransitionTime: metav1.Time{Time: time.Now()},
	}
}
```

and we use like this:
```
podreadiness.SetCondition(condChan, podreadiness.TopologyUpdated, status)
...
podreadiness.SetCondition(condChan, podreadiness.TopologyUpdated, condStatus)
```

This solves my concern and it's not a big shakeup of the current architecture